src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1035,19 +1057,21 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
                                               parameters=mc,
                                               custom_headers=headers).result()
 
-            if enable_managed_identity and attach_acr:
-                # Attach ACR to cluster enabled managed identity
-                if created_cluster.identity_profile is None or \
-                   created_cluster.identity_profile["kubeletidentity"] is None:
-                    logger.warning('Your cluster is successfully created, but we failed to attach acr to it, '
-                                   'you can manually grant permission to the identity named <ClUSTER_NAME>-agentpool '
-                                   'in MC_ resource group to give it permission to pull from ACR.')
-                else:
-                    kubelet_identity_client_id = created_cluster.identity_profile["kubeletidentity"].client_id
-                    _ensure_aks_acr(cmd.cli_ctx,
-                                    client_id=kubelet_identity_client_id,
-                                    acr_name_or_id=attach_acr,
-                                    subscription_id=subscription_id)
+            if enable_managed_identity:
+                if attach_acr:
```

Comment:
is there any special consideration here to split original one line condition check into two ?

Reply:
Hi @akshaysngupta , do we need to merge this change?